src/transformers/models/led/modeling_tf_led.py

```diff
@@ -65,11 +64,12 @@ def shift_tokens_right(input_ids: tf.Tensor, pad_token_id: int, decoder_start_to
     )
 
     # "Verify that `labels` has only positive values and -100"
-    assert_gte0 = tf.debugging.assert_greater_equal(shifted_input_ids, tf.cast(0, tf.int32))
+    if tf.executing_eagerly():
+        assert_gte0 = tf.debugging.assert_greater_equal(shifted_input_ids, tf.constant(0))
 
-    # Make sure the assertion op is called by wrapping the result in an identity no-op
-    with tf.control_dependencies([assert_gte0]):
-        shifted_input_ids = tf.identity(shifted_input_ids)
+        # Make sure the assertion op is called by wrapping the result in an identity no-op
+        with tf.control_dependencies([assert_gte0]):
+            shifted_input_ids = tf.identity(shifted_input_ids)
```

Comment:
What are those two lines for, and why do they need to only be in eager mode?

Reply:
Those are here to be aligned with what looks like the method in BART-like models. `tf.control_dependencies` makes sure that the assert is run in proper order.  Nevertheless, `tf.control_depencies` is useless in TF2 and can safely be removed. This will be the case when I will clean the models as I did for ALBERT and BERT.